### PR TITLE
Allow remove splash screen

### PR DESCRIPTION
**Pull request by fisx**
https://wearezeta.atlassian.net/browse/SQSERVICES-759

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] If this PR changes development workflow or dependencies, they have been A) automated and B) documented under docs/developer/. All efforts have been taken to minimize development setup breakage or slowdown for co-workers.
 - [x] If HTTP endpoint paths have been added or renamed, or feature configs have changed, the **endpoint / config-flag checklist** (see Wire-employee only backend [wiki page](https://github.com/zinfra/backend-wiki/wiki/Checklists)) has been followed.
 - [x] If a cassandra schema migration has been added, I ran **`make git-add-cassandra-schema`** to update the cassandra schema documentation.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.
   - [x] If new config options introduced: added usage description under docs/reference/config-options.md
   - [x] If new config options introduced: recommended measures to be taken by on-premise instance operators.
   - [x] If a cassandra schema migration is backwards incompatible (see also [these docs](https://github.com/wireapp/wire-server/blob/develop/docs/developer/cassandra-interaction.md#cassandra-schema-migrations)), measures to be taken by on-premise instance operators are explained.
   - [x] If a data migration (not schema migration) introduced: measures to be taken by on-premise instance operators.
   - [x] If public end-points have been changed or added: does nginz need un upgrade?
   - [x] If internal end-points have been added or changed: which services have to be deployed in a specific order?

### CODE REVIEW

File: services/galley/src/Galley/API/Teams.hs [hunk 1]
```diff
@@ -336,6 +337,29 @@ updateTeamH zusr zcon tid updateData = do
...
+  ConnId ->
+  TeamId ->
+  Sem r ()
+updateTeamDeleteSplashScreenH zusr zcon tid = do
+  zusrMembership <- E.getTeamMember tid zusr
+  void $ permissionCheckS SSetTeamData zusrMembership
+  E.deleteTeamSplashScreen tid
+  now <- input
+  memList <- getTeamMembersForFanout tid
+  let e = newEvent tid now (EdTeamUpdate newTeamUpdateData)
```
Comment (fisx) [inline services/galley/src/Galley/API/Teams.hs:359]: I'm not very happy about this: deleting the splash screen is not the same as changing nothing.  :)  Maybe add a new constructor to `TeamUpdateData`?

File: services/galley/src/Galley/API/Teams.hs [hunk 2]
```diff
@@ -336,6 +337,29 @@ updateTeamH zusr zcon tid updateData = do
...
+     ]
+    r =>
+  UserId ->
+  ConnId ->
+  TeamId ->
+  Sem r ()
+updateTeamDeleteSplashScreenH zusr zcon tid = do
+  zusrMembership <- E.getTeamMember tid zusr
+  void $ permissionCheckS SSetTeamData zusrMembership
+  E.deleteTeamSplashScreen tid
```
Comment (fisx) [inline services/galley/src/Galley/API/Teams.hs:356]: I'm not convinced this is the right approach, but the alternative of changing `TeamUpdateData` to accomodate both leaving the splash screen intact (`Nothing`) and setting it to `Nothing` (???) doesn't appeal to me either.  We could change the `Maybe` to something like:

```haskell
data ChangeSplashScreen = Keep | Delete | Set AssetKey
```

But then how would that translate into json?  like this?:

```
{ ... 
  "splash_screen": null | {"delete"} | {"set": "key"},
  ...
}
```

Comment (fisx) [reply services/galley/src/Galley/API/Teams.hs:356]: I mean:

```
  "splash_screen": null | {"delete": true} | {"set": "key"},
```

Comment (fisx) [reply services/galley/src/Galley/API/Teams.hs]: I ended up doing what Patryk suggested earlier, namely do the same thing we're already doing for the Icon (aka Logo) [here](https://github.com/wireapp/wire-server/blob/5b0a276051990c4ff51eeff77798ab15715cd059/libs/wire-api/src/Wire/API/Team.hs#L247).

**Comment by fisx** [comment]
Golden tests are still failing, but it's ready for review!

**Comment by fisx** [comment]
ok, this failure is also legit, fix coming up.

**Review by pcapriotti — APPROVED** [review]
LGTM.

**Pull request merged**